src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.Helpers.cs

```diff
@@ -0,0 +1,154 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Extensions
+#if BUILDING_SOURCE_GENERATOR_TESTS
+    .SourceGeneration
+#endif
+    .Configuration.Binder.Tests
+{
+    public static class TestHelpers
+    {
+        public static bool NotSourceGenMode
+#if BUILDING_SOURCE_GENERATOR_TESTS
+            = false;
+#else
+            = true;
+#endif            
+    }
+
+    #region // Shared test classes
+    public class ComplexOptions
```

Comment:
>ComplexOptions [](http://example.com/codeflow?start=17&length=14)

I see you introduced that. I think this is good.

Reply:
😬actually I didn't, just moved from the original root binder tests per https://github.com/dotnet/runtime/pull/82179#discussion_r1141083507.